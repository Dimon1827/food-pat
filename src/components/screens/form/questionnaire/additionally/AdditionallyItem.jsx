import SelectInput from '../../../../ui/selectInput/SelectInput';

function AdditionallyItem({ id, name, type }) {
  return (
    <li className="additionally__item">
      <SelectInput id={id} type={type} name={name} />
    </li>
  );
}

export default AdditionallyItem;
