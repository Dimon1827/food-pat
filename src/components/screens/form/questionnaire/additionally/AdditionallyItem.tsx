import { ICheckBoxItem } from '@/types';
import SelectInput from '@/components/ui/selectInput/SelectInput';

function AdditionallyItem(item: ICheckBoxItem) {
  return (
    <li className="additionally__item">
      <SelectInput {...item} />
    </li>
  );
}

export default AdditionallyItem;
