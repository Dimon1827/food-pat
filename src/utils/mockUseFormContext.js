import { useFormContext } from "react-hook-form";

export function mockUseFormContext(errors) {
  useFormContext.mockReturnValue({
    register: jest.fn(),
    formState: {
      errors: errors,
    },
  });
}


