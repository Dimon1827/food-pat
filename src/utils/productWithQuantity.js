const productWithQuantity = (food) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
