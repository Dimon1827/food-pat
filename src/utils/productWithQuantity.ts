import { IProducts } from "redux/slice/productsSlice.types";

interface IproductWithQuantity extends Omit<IProducts, 'isChecked' | 'quantity'> {};

const productWithQuantity = (food: IproductWithQuantity) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
