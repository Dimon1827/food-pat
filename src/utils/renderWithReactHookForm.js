import { render } from '@testing-library/react';
import { FormProvider, useForm } from 'react-hook-form';

export function renderWithReactHookForm(ui, { defaultValues = {} } = {}) {
  let reactHookFormMethods = {};

  const Wrapper = ({ children }) => {
    const methods = useForm({ defaultValues });
    return <FormProvider {...methods}>{children}</FormProvider>;
  };

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}