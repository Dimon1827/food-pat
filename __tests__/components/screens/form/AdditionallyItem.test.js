import { screen } from '@testing-library/react';
import { renderWithReactHookForm } from '@/utils/renderWithReactHookForm';
import AdditionallyItem from '@/components/screens/form/questionnaire/additionally/AdditionallyItem';

const inputElement = {
  id: 'milk',
  type: 'checkbox',
  name: 'Молоко',
};

describe('AdditionallyItem', () => {
  it('AdditionallyItem renders correctly as checkbox', () => {
    renderWithReactHookForm(<AdditionallyItem {...inputElement} />);
    const input = screen.getByRole('checkbox');
    const listItem = screen.getByRole('listitem');
    const label = screen.getByText('Молоко');
    expect(listItem).toMatchSnapshot();
    expect(input).toBeInTheDocument();
    expect(label).toBeInTheDocument();
    expect(input).toHaveClass('additionally__input');
    expect(label).toHaveClass('additionally__label');
  });
});
