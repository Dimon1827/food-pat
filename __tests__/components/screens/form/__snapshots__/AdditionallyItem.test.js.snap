// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionallyItem AdditionallyItem renders correctly as checkbox 1`] = `
<li
  class="additionally__item"
>
  <input
    class="visually-hidden control-input additionally__input"
    id="milk"
    name="milk"
    type="checkbox"
    value="milk"
  />
  <label
    class="label additionally__label"
    for="milk"
  >
    Молоко
  </label>
</li>
`;
